pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -422,15 +422,77 @@ func setNodeStatusFn(status *operatorv1.NodeStatus) v1helpers.UpdateStaticPodSta
 	}
 }
 
+type revisionDescriptionPrinter struct {
+	nodes          []operatorv1.NodeStatus
+	latestRevision int32
+	progressing    bool
+}
+
+func newRevisionDescriptionPrinter(nodes []operatorv1.NodeStatus, latestRevision int32, progressing bool) *revisionDescriptionPrinter {
+	return &revisionDescriptionPrinter{nodes: nodes, latestRevision: latestRevision, progressing: progressing}
+}
+
+func pluralizedNode(i int) string {
+	if i < 2 {
+		return "node"
+	}
+	return "nodes"
+}
+
+func (r revisionDescriptionPrinter) String() string {
+	currentRevisions := sets.NewInt32()
+	latestCount := 0
+	availableCount := 0
+	for i := range r.nodes {
+		c := r.nodes[i].CurrentRevision
+		currentRevisions.Insert(c)
+		if c == r.latestRevision {
+			latestCount++
+		}
+		if c != 0 {
+			availableCount++
```

Comment:
why does this mean anything about availability?

Reply:
in the existing code "numAvailable" is determined by CurrentRevision!=0 ... So i guess CurrentRevision==0 means the node is not available or not actively making progress? i'm a bit confused here as well.

Reply:
this has nothing to do with availability. It only says that this revision was the last that got ready eventually.

Reply:
`lastReadyRevision` ?

Reply:
call it `nonZeroCount`

Reply:
> lastReadyRevision

It's a count.